### PR TITLE
feat(commerce): create PLP v2 interactive result controller

### PR DESCRIPTION
**Pull request by Spuffynism**
The implementation is effectively a copy of the [PLP v1 implementation](https://github.com/coveo/ui-kit/tree/master/packages/headless/src/controllers/product-listing/result-list).
[CAPI-85]

[CAPI-85]: https://coveord.atlassian.net/browse/CAPI-85?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/headless/src/controllers/product-listing-v2/result-list/headless-product-listing-v2-interactive-result.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+export interface InteractiveResultProps extends InteractiveResultCoreProps {
+  /**
+   * The options for the `InteractiveResult` controller.
+   * */
+  options: InteractiveResultOptions;
+}
+
+/**
+ * The `InteractiveResult` controller provides an interface for triggering desirable side effects, such as logging UA events to the Coveo Platform, when a user selects a query result.
```
Comment (fbeaudoincoveo) [inline packages/headless/src/controllers/product-listing-v2/result-list/headless-product-listing-v2-interactive-result.ts:27]:
```suggestion
 * The `InteractiveResult` controller provides an interface for triggering desirable side effects, such as logging UA events to the Coveo Platform, when a user selects a product.
```

File: packages/headless/src/controllers/product-listing-v2/result-list/headless-product-listing-v2-interactive-result.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+ */
+export interface InteractiveResult extends InteractiveResultCore {}
+
+/**
+ * Creates an `InteractiveResult` controller instance.
+ *
+ * @param engine - The headless engine.
```
Comment (fbeaudoincoveo) [inline packages/headless/src/controllers/product-listing-v2/result-list/headless-product-listing-v2-interactive-result.ts:34]:
```suggestion
 * @param engine - The headless commerce engine.
```

File: packages/headless/src/controllers/product-listing-v2/result-list/headless-product-listing-v2-interactive-result.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import {
+  buildInteractiveResultCore,
+  InteractiveResultCore,
+  InteractiveResultCoreOptions,
+  InteractiveResultCoreProps,
+} from '../../core/interactive-result/headless-core-interactive-result';
+
+export interface InteractiveResultOptions extends InteractiveResultCoreOptions {
+  /**
+   * The query result.
```
Comment (fbeaudoincoveo) [inline packages/headless/src/controllers/product-listing-v2/result-list/headless-product-listing-v2-interactive-result.ts:14]:
```suggestion
   * The product.
```

File: packages/headless/src/controllers/product-listing-v2/result-list/headless-product-listing-v2-interactive-result.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
+import {ProductRecommendation} from '../../../api/search/search/product-recommendation';
```
Comment (fbeaudoincoveo) [inline packages/headless/src/controllers/product-listing-v2/result-list/headless-product-listing-v2-interactive-result.ts:1]: We could potentially rename the controller to "InteractiveProduct" or something.

But technically, products are query results, so I'm not opposed to using the generic name.

File: packages/headless/src/controllers/product-listing-v2/result-list/headless-product-listing-v2-interactive-result.ts (hunk 1 above)
Comment (fbeaudoincoveo) [reply packages/headless/src/controllers/product-listing-v2/result-list/headless-product-listing-v2-interactive-result.ts:27]: This makes me wonder what other desirable side effects this controller could ever trigger...

Comment (fbeaudoincoveo) [reply packages/headless/src/controllers/product-listing-v2/result-list/headless-product-listing-v2-interactive-result.ts:27]: I just discussed it with @olamothe and I now see that it does other stuff under the hood such as handling long presses on mobile, or multiple clicks

Here's an updated suggestion for the documentation:

```
The `InteractiveProduct` controller provides an interface for handling long presses, multiple clicks, etc. to ensure that Coveo usage analytics events are logged properly when a user selects a product.
```

**Review by dmbrooke — APPROVED** [review]

**Review by btaillon — APPROVED** [review]

**Pull request merged**